.github/workflows/dependency_check.yaml

```diff
@@ -0,0 +1,49 @@
+name: Check For Unused Dependencies
```

Comment:
I think it might be good to explicitly scope this to the frontend. So maybe change the name of the yaml to `dependency_check_frontend.yaml` and modify the name to `Check for Unused Frontend Dependencies`?